pkg/ddc/jindo/shutdown.go

```diff
@@ -129,11 +159,38 @@ func (e *JindoEngine) destroyWorkers(workers int32) (err error) {
 		if len(toUpdate.Labels) < len(node.Labels) {
 			err := e.Client.Update(context.TODO(), toUpdate)
 			if err != nil {
-				return err
+				return expectedWorkers, err
 			}
-			e.Log.Info("Destory worker", "Dataset", e.name, "deleted worker node", node.Name, "removed labels", labelNames)
+			e.Log.Info("Destroy worker", "Dataset", e.name, "deleted worker node", node.Name, "removed labels", labelNames)
 		}
+
+		currentWorkers--
 	}
 
-	return
+	return currentWorkers, nil
+}
+
+func (e *JindoEngine) sortNodesToShutdown(candidateNodes []corev1.Node, fuseGlobal bool) (nodes []corev1.Node, err error) {
+	if !fuseGlobal {
+		// If fuses are deployed in non-global mode, workers and fuses will be scaled in together.
+		// It can be dangerous if we scale in nodes where there are pods using the related pvc.
+		// So firstly we filter out such nodes
+		pvcMountNodes, err := kubeclient.GetPvcMountNodes(e.Client, e.name, e.namespace)
+		if err != nil {
+			e.Log.Error(err, "GetPvcMountNodes when scaling in")
+			return nil, err
+		}
+
+		for _, node := range candidateNodes {
+			if _, found := pvcMountNodes[node.Name]; !found {
+				nodes = append(nodes, node)
+			}
+		}
+	} else {
+		// If fuses are deployed in global mode. Scaling in workers has nothing to do with fuses.
```

Comment:
Even if scaling in workers has nothing to do with fuses, it's better to scale workers without PVC mount Pods first.

Reply:
jindoruntime dont't supoort fuse global option, so that's will implentment and test by next time